### PR TITLE
Open and check

### PR DESCRIPTION
**Pull request by Antonio95**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (master)
- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: src/ligero/mod.rs [hunk 1]
```diff
@@ -469,7 +490,73 @@ where
...
+            MerkleTree::<C>::new(&leaf_hash_param, &two_to_one_param, col_hashes).unwrap();
+
+        // 5. Compute Merkle tree paths for the columns
+        let mut queried_columns = Vec::new();
+        let mut paths = Vec::new();
+
+        for i in indices {
+            queried_columns.push(ext_mat_cols[i].clone());
+            paths.push(col_tree.generate_proof(i).unwrap());
+        }
```
Comment (diegode) [inline src/ligero/mod.rs:552]: This block is a copy of commit(); we should move it into a function

File: src/ligero/mod.rs [hunk 2]
```diff
@@ -354,6 +373,9 @@ where
     {
         // 0. Recovering parameters
         let t = calculate_t(rho_inv, sec_param);
+        let mut optional = crate::optional_rng::OptionalRng(rng); // TODO taken from Marlin code; change in the future?
+        let leaf_hash_param = C::LeafHash::setup(&mut optional).unwrap();
```
Comment (mmagician) [inline src/ligero/mod.rs:377]: should be from vk/ck

File: src/ligero/mod.rs [hunk 3]
```diff
@@ -469,7 +490,73 @@ where
...
         Self::Commitment: 'a,
     {
-        todo!()
+        let labeled_polynomial = labeled_polynomials.into_iter().next().unwrap();
+        let polynomial = labeled_polynomial.polynomial();
+        let num_elems = polynomial.degree() + 1;
+        let m = (num_elems as f64).sqrt().ceil() as usize;
+        let t = calculate_t(rho_inv, sec_param);
+        let mut optional = crate::optional_rng::OptionalRng(rng);
+        let leaf_hash_param = C::LeafHash::setup(&mut optional).unwrap();
```
Comment (mmagician) [inline src/ligero/mod.rs:499]: same here

File: src/ligero/mod.rs [hunk 4]
```diff
@@ -485,13 +572,55 @@ where
...
+            t,
+            &mut transcript,
+            &vk.leaf_hash_params,
+            &vk.two_to_one_params
+        )?;
+
+        return Ok(inner_product(&commitment.proof.v, &a) == values.into_iter().next().unwrap());
+
+        // TODO what is "values"? does it contain the actual evaluations?
         todo!()
```
Comment (diegode) [inline src/ligero/mod.rs:624]: We can remove this "todo"

**Review by mmagician — APPROVED** [review]

**Pull request merged**